wordpress/wp-content/plugins/sil-dictionary-webonary/include/searchform_func.php

```diff
@@ -24,6 +24,63 @@ function webonary_searchform() {
 	{
 		return false;
 	}
+
+	/*
+	 * Set up the Parts of Speech
+	 */
+	$taxonomy = filter_input(INPUT_GET, 'tax', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+	$parts_of_speech_dropdown = '';
+
+	if(get_option('useCloudBackend'))
+	{
+		$arrIndexed = array();
+		$dictionaryId = Webonary_Cloud::getBlogDictionaryId();
+		$dictionary = Webonary_Cloud::getDictionary($dictionaryId);
+		if(!is_null($dictionary))
+		{
+			if (count($dictionary->mainLanguage->partsOfSpeech))
+			{
+				$parts_of_speech_dropdown .= "<select  name='tax' id='tax' class='postform' >";
+				$parts_of_speech_dropdown .= "<option value=''>" . __('All Parts of Speech','sil_dictionary') . "</option>";
+				foreach ($dictionary->mainLanguage->partsOfSpeech as $part)
+				{
+					$selected = ($part === $taxonomy) ? ' selected ' : '';
+					$parts_of_speech_dropdown .= "<option value=" . $part . $selected . ">" . $part . "</option>";
+				}
+				$parts_of_speech_dropdown .= "</select>";
+			}
+
+			$indexed = new stdClass();
+			$indexed->language_name = $dictionary->mainLanguage->title;
+			$indexed->totalIndexed = $dictionary->mainLanguage->entriesCount ?? 0;
+			$arrIndexed[] = $indexed;
+			foreach($dictionary->reversalLanguages as $index => $reversal)
+			{
+				$indexed = new stdClass();
+				$indexed->language_name = $reversal->title;
+				$indexed->totalIndexed = $reversal->entriesCount ?? 0;
+				$arrIndex[] = $indexed;
+			}
+		}
+
+		$lastEditDate = $dictionary->updatedAt;
```

Comment:
This will throw an exception if `$dictionary` is null.

Reply:
initialized, and set inside the !is_null($dictionary) block.